refarch/aws-native/common/common_cdk/audit_trail_glue.py

```diff
@@ -17,7 +17,11 @@ def __init__(self, scope: Construct, id: str, audit_db: Database, audit_table: s
         super().__init__(scope, id, **kwargs)
 
         # CloudTrail - S3 event Logging for the requested bucket
-        self.__trail = Trail(self, 'AuditTrail', bucket=log_bucket)
+        # CloudFormation has a bug that put double "--" in the trail name
+        # This violates the naming constraint of cloudtrail
+        self.__trail = Trail(self, 'AuditTrail', 
+            bucket=log_bucket,
+            trail_name=f'{id}-{Stack.of(self).stack_name}-{Stack.of(self).account}')
```

Comment:
This bug only happens when you deploy cross-account. 

I checked that the generated template is the same whether you deploy with CICD or directly.

However, when using CICD, the stack name changes from `ara-<module>` to `AnalyticsPipelineStage-ara-<module>`. The length may trigger this bug. Since it's in CloudFormation level, I think the best fix is to specify the `trail_name` that is unique enough by ourselves.